### PR TITLE
add minor release doc updates

### PR DESCRIPTION
**Pull request by justin1121**

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ Run the following to actually upload to cargo:
 cargo publish
 ```
 
+Next you need to manually make the release in github from the tag. This will kick off the building process
+to build all the releases assets and store them on the release in github.
```
Comment (jvmncs) [inline RELEASING.md:16]: would mention something here about pressing the button to auto-generate release notes, noticed we've forgotten to do that a couple times

**Review by jvmncs — APPROVED** [review]
lgtm w/ comment

File: RELEASING.md (hunk 1 above)
Comment (justin1121) [reply RELEASING.md:16]: Oh I think I didn't even know this was an option. How would you do that?

Comment (jvmncs) [reply RELEASING.md:16]: It's a button in the GitHub UI when you're creating a release. If you try to edit any of the existing releases you'll see it above the Release Notes text box

**Pull request merged**